dex/farm/src/lib.rs

```diff
@@ -123,9 +123,9 @@ pub trait Farm:
         require!(self.is_active(), "Not active");
         require!(!self.farm_token_id().is_empty(), "No farm token");
 
-        let payments = self.get_all_payments_managed_vec();
-        require!(payments.len() >= 1, "empty payments");
-        let payment_0 = payments.get(0).unwrap();
+        let payments_vec = self.get_all_payments_managed_vec();
+        let mut payments = payments_vec.iter();
+        let payment_0 = payments.next().ok_or("empty payments")?;
```

Comment:
`first_payment` maybe? It's a lot easier to make a mistake with numbers in the name.